### PR TITLE
prevent sign extension of char to int for isprint

### PR DESCRIPTION
**Pull request by mitza-oci**

### CODE REVIEW

File: dds/idl/dds_generator.h [hunk 1]
```diff
@@ -661,7 +661,7 @@ std::ostream& char_helper(std::ostream& o, CharType value)
   case '\a':
     return o << "\\a";
   }
-  if (isprint(value)) {
+  if (isprint(static_cast<unsigned char>(value))) {
```
Comment (jwillemsen) [inline dds/idl/dds_generator.h:664]: Why not use ACE_OS::ace_isprint?

Comment (mitza-oci) [reply dds/idl/dds_generator.h:664]: I didn't write the code, so I'll let @iguessthislldo answer specifically.  I'm just trying to get the builds fixed.
There is a `C++ Standard` section in guidelines.md that is relevant.

Comment (iguessthislldo) [reply dds/idl/dds_generator.h:664]: This needs to work for `wchar` too, so maybe it needs to use `char_value`, which I defined right about this funcion?

Comment (jwillemsen) [reply dds/idl/dds_generator.h:664]: is_print result depends on the locale, different locale could result in different output, is that intended?

Comment (iguessthislldo) [reply dds/idl/dds_generator.h:664]: Also what ever we do, it should not truncate the `wchar_t` value like this looks like it would.

Comment (mitza-oci) [reply dds/idl/dds_generator.h:664]: `wchar_t` has its own stdlib function (`iswprint`?)

Comment (iguessthislldo) [reply dds/idl/dds_generator.h:664]: Yes, but we can't print a directly print a `wchar_t` to the byte-char based `ostream`. We have to basically see if it's ASCII first, else it falls back to the hex value.

Comment (iguessthislldo) [reply dds/idl/dds_generator.h]: > is_print result depends on the locale, different locale could result in different output, is that intended

Sorta. I'm not really at ease with how text encodings are handled (or more accurately not) in IDL and DDS in general, but I figured it was good enough for now. I was just trying to improve it so it's not just printing out the char as is or the wchar casted to a char, but easier to read than doing something like always using a hex escape.

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**